core/block_validator.go

```diff
@@ -65,6 +66,11 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash {
 		return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash)
 	}
+	if v.config.IsShanghai(new(big.Int).SetUint64(block.Time())) {
+		if hash := types.DeriveSha(block.Withdrawals(), trie.NewStackTrie(nil)); hash != *header.WithdrawalsHash {
```

Comment:
Is it possible that an invalid block is given and the withdrawalsHash is nil? Perhaps check the nilness here to prevent panic?

Reply:
I agree with @rjl493456442